tensorflow/compiler/xla/service/gpu/BUILD

```diff
@@ -907,93 +991,36 @@ cc_library(
     alwayslink = True,  # Contains per-platform transfer manager registration
 )
 
+cc_library(
+    name = "gpu_compiler",
+    deps = if_cuda_is_configured([":nvptx_compiler"])
+          + if_rocm_is_configured([":amdgpu_compiler"]),
```

Comment:
I know that TensorFlow has a different philosophy, but inside of XLA I'm going to ask that we build a system such that one XLA binary can compile for both AMD and nvidia.  XLA does not require a CUDA compiler at all, so this is totally possible as far as XLA is concerned.

I understand that you all have designed StreamExecutor in a way which precludes this use-case.  That's unfortunate.  But I think we need to start making things right now, rather than compounding on this decision.

The benefits of having one compiler that can target both nvidia and AMD will be very large for our customers such as JAX, who ship precompiled binaries.  It will also make testing inside of google much easier and so will make it much more likely that we can turn on presubmits when the time comes; I know that's something you all are concerned with.